tt_eager/tensor/types.hpp

```diff
@@ -449,7 +449,8 @@ struct MultiDeviceHostStorage {
         std::vector<int> ordered_device_ids;
         std::unordered_map<int, DeviceBuffer> buffers;
         std::unordered_map<int, Shape> shapes;
-        mutable std::mutex mtx;
+        mutable std::mutex buffer_mtx;
```

Comment:
Curious how much did this change improved e2e perf?

Reply:
~2-3% ... which seems like noise but it seemed consistently better across runs.